File: app-core/src/main/java/code/sdk/core/Constant.java
package code.sdk.core;

import androidx.annotation.StringDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

public class Constant {
    public static final String TAG = Constant.class.getSimpleName();

    /* target country iso code (Alpha-2 lowcase) */
    public static final String TARGET_COUNTRY_INDIA = "in";
    public static final String TARGET_COUNTRY_BRAZIL = "br";
    public static final String TARGET_COUNTRY_INDONESIA = "id";
    public static final String TARGET_COUNTRY_VIETNAM = "vn";

    public static final long DELAY_FOR_DEVICE_ID_ASYNC = 1500L;


    /* url query params */
    public static final String QUERY_PARAM_ORIENTATION = "orientation";
    public static final String QUERY_PARAM_HOVER_MENU = "hoverMenu";
    public static final String QUERY_PARAM_NAV_BAR = "navBar";
    public static final String QUERY_PARAM_SAFE_CUTOUT = "safeCutout";

    /* screen orientation */
    public static final String LANDSCAPE = "landscape";
    public static final String PORTRAIT = "portrait";
    public static final String UNSPECIFIED = "unspecified";

    @StringDef({PORTRAIT, LANDSCAPE, UNSPECIFIED})
    @Retention(RetentionPolicy.SOURCE)
    public @interface ScreenOrientation {
    }

    /* hover menu */
    public static final String DOCK_LEFT = "dockLeft";
    public static final String DOCK_RIGHT = "dockRight";

    @StringDef({DOCK_LEFT, DOCK_RIGHT})
    @Retention(RetentionPolicy.SOURCE)
    public @interface HoverMenuDockType {
    }

    public static final String STATE_DOCKED = "stateDocked";
    public static final String STATE_DRAGGING = "stateDragging";
    public static final String STATE_EXPANDED = "stateExpanded";

    @StringDef({STATE_DOCKED, STATE_DRAGGING, STATE_EXPANDED})
    @Retention(RetentionPolicy.SOURCE)
    public @interface HoverMenuState {
    }

}

File: app-core/src/main/java/code/sdk/core/Constant.kt
package code.sdk.core

import androidx.annotation.StringDef

object Constant {
   private val TAG = Constant::class.java.simpleName

    /* target country iso code (Alpha-2 lowcase) */
    const val TARGET_COUNTRY_INDIA = "in"
    const val TARGET_COUNTRY_BRAZIL = "br"
    const val TARGET_COUNTRY_INDONESIA = "id"
    const val TARGET_COUNTRY_VIETNAM = "vn"
    const val DELAY_FOR_DEVICE_ID_ASYNC = 1500L
    const val SELF = 200
    const val OTHER_CODE = 1

    /* url query params */
    const val QUERY_PARAM_ORIENTATION = "orientation"
    const val QUERY_PARAM_HOVER_MENU = "hoverMenu"
    const val QUERY_PARAM_NAV_BAR = "navBar"
    const val QUERY_PARAM_SAFE_CUTOUT = "safeCutout"

    /* screen orientation */
    const val LANDSCAPE = "landscape"
    const val PORTRAIT = "portrait"
    const val UNSPECIFIED = "unspecified"

    /* hover menu */
    const val DOCK_LEFT = "dockLeft"
    const val DOCK_RIGHT = "dockRight"
    const val STATE_DOCKED = "stateDocked"
    const val STATE_DRAGGING = "stateDragging"
    const val STATE_EXPANDED = "stateExpanded"

    @StringDef(PORTRAIT, LANDSCAPE, UNSPECIFIED)
    @Retention(AnnotationRetention.SOURCE)
    annotation class ScreenOrientation

    @StringDef(DOCK_LEFT, DOCK_RIGHT)
    @Retention(AnnotationRetention.SOURCE)
    annotation class HoverMenuDockType

    @StringDef(STATE_DOCKED, STATE_DRAGGING, STATE_EXPANDED)
    @Retention(AnnotationRetention.SOURCE)
    annotation class HoverMenuState
}
